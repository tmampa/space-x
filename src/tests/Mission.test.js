import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../../src/redux/configureStore";
import MissionsTable from "../../src/components/missionsComponents/MissionsTable";

describe("Mission Element Tests", () => {
	test("Check if Component Renders in DOM", () => {
		render(
			<Provider store={store}>
				<MissionsTable />
			</Provider>
		);
		expect(screen.queryByText(/Rocket/)).toBeNull();
		expect(screen.queryByText(/Mission/)).toBeTruthy();
	});
	it("renders correctly", () => {
		const tree = renderer
			.create(
				<Provider store={store}>
					<MissionsTable />
				</Provider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
