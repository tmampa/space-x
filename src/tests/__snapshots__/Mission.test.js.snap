// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission Element Tests renders correctly 1`] = `
<table>
  <tbody>
    <tr>
      <th>
        Mission
      </th>
      <th>
        Description
      </th>
      <th>
        Status
      </th>
      <th>
         
      </th>
    </tr>
  </tbody>
</table>
`;
